test/stablehlo/test_mark_pattern.py

```diff
@@ -0,0 +1,144 @@
+import sys
+import unittest
+
+import torch
+import torch_xla.core.xla_model as xm
+import torch_xla.experimental.xla_marker
+from torch.utils import _pytree as pytree
+
+
+class XlaMarkPatternTest(unittest.TestCase):
+
+  def run_func_get_stablehlo(self, f, input_args):
+
+    device = xm.xla_device()
+    input_args = pytree.tree_map_only(torch.Tensor,
+                                      lambda x: x.to(device=device), input_args)
+    out = f(*input_args)
+    if isinstance(out, tuple):
+      out = list(out)
+    else:
+      out = [out]
+    stablehlo = xm.get_stablehlo(out)
+    return stablehlo
+
+  def test_basic(self):
+
+    def f(x):
+      x = x + 1
+      x = torch.ops.xla_pattern_marking.mark_tensor(x, "p", 0, 0, True)
```

Comment:
4. In an ideal world we wouldn't need `id` either. The algorithm could be: Find the first `mark_end` custom_call and traverse the use-def chain upwards until we hit the first `mark_begin` with matching name. This would handle lexical scoping of nested abstractions without the need for IDs

Reply:
I like these suggestions. To be clear, we can still keep the one MarkTensor Lazy IR node if we want; but the exposed Python API can be mark_begin / mark_end

Reply:
> Could remove the need for tracking position via `pos=`

I think it is okay to keep the low level API here to be raw. There can be some tricky requirements from different HLFB frontends e.g. pattern matching requires inserting a additional mark tensor node to the FX graph for each input and output [ref](https://github.com/arfaian/odml-sdk/blob/2f08c60d1d33397614de4edff60eb45cce9b2a99/odml_sdk/experimental/torch/hlfb/mark_pattern.py#L180).  ODML SDK will definitely build additional HLFB layers to wrap the torch_xla low level APIs for different frontends and safety guards.


> In an ideal world we wouldn't need id either

I'd lean toward keeping the id to guarantee the correctness of composites from MLIR pass. TBH I also want an additional field `total_inputs` and `total_outputs` to make sure all the boundary IO are correctly identified during the graph traversal.